packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ProxyEOA.sol

```diff
@@ -62,23 +63,27 @@ contract OVM_ProxyEOA {
      * Public Functions *
      ********************/
 
-    /**
-     * Changes the implementation address.
-     * @param _implementation New implementation address.
-     */
-    function upgrade(
-        address _implementation
-    )
-        external
-    {
-        require(
-            msg.sender == Lib_ExecutionManagerWrapper.ovmADDRESS(),
-            "EOAs can only upgrade their own EOA implementation"
-        );
-
-        _setImplementation(_implementation);
-        emit Upgraded(_implementation);
-    }
+    // NOTE: Upgrades are temporarily disabled while we construct a whitelist-based system of
```

Comment:
I don't think we should ever do a whitelist system of smart contract accounts because that is inherently not scalable. You should update this comment so that it doesn't mention a particular solution